File: app/src/main/java/com/schrondinger/quin/base/activity/ActivityInject.java
package com.schrondinger.quin.base.activity;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Schrodinger on 2017/12/28.
 */

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ActivityInject {
    int rootViewId() default -1;
    int title() default -1;
}

File: app/src/main/java/com/schrondinger/quin/base/activity/ActivityInject.kt
package com.schrondinger.quin.base.activity

/**
 * ================================================
 * 作    者：schrodinger
 * 版    本：2.0
 * 创建日期： 2017/12/28 9:29 AM
 * 描    述：
 * 修订历史：2018/12/17 9:29 AM
 * ================================================
 */
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class ActivityInject(val rootViewId: Int = -1, val title: Int = -1)
